### ISSUE TITLE
Which input source should be selected as microphone

### ISSUE THREAD

**Issue by Adelenade** [Adelenade/GoXlr-Macos#11]
### Moved onto a new issue

Which input source should we be selecting for microphone? 'GoXLR'?

_Originally posted by @brymon68 in https://github.com/Adelenade/GoXlr-Macos/issues/9#issuecomment-1189264800_

I'm going to do a post about that, here, Just wait a minute !

**Comment by Adelenade** [comment]
Okay so, little explanation of the inputs of the GoXLR (chat mic / broadcast mix)

### How the system see the GoXLR

For MacOS, the goxlr is a multi-channel usb audio interface. Every device you connect to your computer is seen by the system as one, two or more channels. For example, headphones you connect to the jack port of your mac is - if it's a modern one - composed of TWO channels, one for the left ear, one for the right ear. Here, the goXlr have two channels per "sub device" (like chat, system, broadcast-mix etc), one for each "ear". 
So, what can you see when inspecting the inputs (mics) of the device ?

![image](https://user-images.githubusercontent.com/63637808/179806924-fb7aabaa-6ff2-4630-bd44-a99002111bac.png)

Here, you can recognise some names: Chat mic and Broadcast Stream Mix. Each one composed of 2 channels, one L and one R. You also see lots of other channels that doesn't really matter as they are supposed to be hidden from the user.

The ultra-large majority of apps will only use two of theses channels (one for left one for right), and don't include any way to select specific channels as most of the users will only have devices with two channels.

So, to answer your question. You need to select the two channels corresponding to your mic as your mic. As most of the apps just suggest devices and not any way to pick channels from theses devices, you need to use a third party app to split the channels as the goxlr macos app can't split theses channels (I need to code a custom driver and i'm not as good as needed). You can use for example loopback from (again) RogueAmoeba.

BUT, as the system pick the two first channels of the device as the device itself for apps that doesn't care about selecting channels, you can just pick goXlr as your microphone, and adapt the router to have only your mic on broadcast-mix as the system will select the channels corresponding to broadcast mix.

SO, to summarise: 

#### GoXLR device = Broadcast mix

: )

**Comment by brymon68** [comment]
Hmm okay. Does this seem like the correct setup? Attached is a screenshot of sound preferences, Goxlr Panel, SoundSource and Midi Setup: 

![Screen Shot 2022-07-19 at 11 55 12 AM](https://user-images.githubusercontent.com/8397405/179817494-a8dc71ca-1f52-4d3c-bca8-81d61f555b21.png)


I see in the MacOS Sound preferences panel under the 'Input' tab, that my input level is being recognized when I talk but unfortunately my apps dont seem to be picking it up.

**Comment by Adelenade** [comment]
Normally you have to deselect all the chat, music, game etc from stream (as stream = goxlr device) and it will be O.K !
But you may have problems with the gate & the mic processing :3

**Comment by brymon68** [comment]
That worked @Adelenade . Thank you again so much, had to mess with the gate a bit but I'm very happy. Thank you again for your hard work and looking forward to v2

**Comment by Adelenade** [comment]
You're welcome ! I'm closing the issue, feel free to re-open it if needed !
